Cargo.toml

```diff
@@ -1,27 +1,26 @@
 [package]
 name = "esp32-hal"
 version = "0.1.0"
-authors = ["Scott Mabin <scott@mabez.dev>"]
+authors = ["Scott Mabin <scott@mabez.dev>", "Arjan Mels <arjan@mels.email>"]
 edition = "2018"
 
 [dependencies]
-esp32 = { version = "0.2.2" }
-
-[dependencies.embedded-hal]
-features = ["unproven"]
-version = "0.2.3"
+xtensa-lx6-rt = { path = "../xtensa-lx6-rt" }
+esp32 = { path = "../esp32"}
```

Comment:
For the rt crate I can use a link to github. However for the esp32 crate this doesn't work as the src directory is not included in the repo. So this would require releasing a new version, I guess. (Which is not so convenient during this early development work. Should we for the time being include the src directory in this repo? Or?)

Reply:
`[patch.crates-io]` can be specified in the `Cargo.toml` to use dependencies locally for development and still _have_ specified a version published to crates.io, when the HAL crate is published it will use the version published on crates.io.

- [See this for example usage](https://doc.rust-lang.org/edition-guide/rust-2018/cargo-and-crates-io/replacing-dependencies-with-patch.html)
- [Detailed documentation](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section)

Reply:
@jeandudey: thanks that is useful for further development.

However it does not solve the problem with the esp32 dependency as the git repo does not contain the actual rust source, but only the files needed to generate it. So either needs to be published to crates.io or we need to include the rust source in the repo.

@MabezDev: How do you want to proceed?

Reply:
You can use `path = "..esp32"` for development, having the source locally.

Reply:
Anyway to publish this crate to crates.io you can't use the git repository as a dependency. An `esp32` version with the necessary changes needs to be published in order for this crate to be published.

Of course, I would prefer that the source code is included in the repo, as that way the versions can be properly tagged.

Reply:
Agree. @MabezDev shall I make a pull request for esp32 including the src directory?

Reply:
src/ used to be tracked, but this made reviewing changes an nightmare. It's much cleaner now, at the cost of not being able to use git links. I am happy to release a new version of `esp32` though, are we sure we've got everything we need?

Reply:
Clear. I have everything for this pull and the multicore pull I am working on. Of course I expect that more changes will come in as people work on other blocks.

Reply:
I've just released `esp32 = "0.3.0"`!

Reply:
@MabezDev Thanks! I notice that the git repo Cargo.toml is still at 0.2.2.
I updated the cargo toml in this pull request. (I will override for local development in .cargo/config.)

I think this resolves this point?